src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,133 @@
+
+                host_task<detail::kernel_name<mklcpu_desc_t>>(cgh, [=]() {
+                    std::int64_t local_err = DFTI_MEMORY_ERROR;
+                    if (config_values.dimensions.size() == 1)
+                        local_err =
+                            DftiCreateDescriptor(handle_obj.get_pointer(), mklcpu_prec, mklcpu_dom,
+                                                 1, config_values.dimensions[0]);
+                    else
+                        local_err = DftiCreateDescriptor(
+                            handle_obj.get_pointer(), mklcpu_prec, mklcpu_dom,
+                            config_values.dimensions.size(), config_values.dimensions.data());
+
+                    set_value(*handle_obj.get_pointer(), config_values);
+                    if (local_err != DFTI_NO_ERROR) {
+                        throw oneapi::mkl::exception("dft", "commit",
+                                                     "DftiCreateDescriptor failed");
+                    }
+
+                    local_err = DftiCommitDescriptor(*handle_obj.get_pointer());
+                    (*status_obj.get_pointer()) = local_err;
+                });
+            })
+            .wait();
+
+        status = status_buffer.template get_access<sycl::access::mode::read>()[0];
         if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCreateDescriptor failed");
+            throw oneapi::mkl::exception("dft", "commit", "DftiCommitDescriptor failed");
         }
 
-        set_value(handle, config_values);
-
-        status = DftiCommitDescriptor(handle);
-        if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCommitDescriptor failed");
-        }
+        device_handle = handle_buffer.template get_access<sycl::access::mode::read>()[0];
     }
 
-    virtual void* get_handle() noexcept override {
-        return handle;
+    void* get_handle() noexcept override {
+        return reinterpret_cast<void*>(device_handle);
     }
 
     virtual ~commit_derived_impl() override {
-        DftiFreeDescriptor((DFTI_DESCRIPTOR_HANDLE*)&handle);
+        DftiFreeDescriptor((mklcpu_desc_t*)&device_handle);
     }
 
 private:
-    DFTI_DESCRIPTOR_HANDLE handle = nullptr;
-
-    constexpr DFTI_CONFIG_VALUE get_domain(domain d) {
-        if (d == domain::COMPLEX) {
-            return DFTI_COMPLEX;
-        }
-        else {
-            return DFTI_REAL;
-        }
-    }
-
-    constexpr DFTI_CONFIG_VALUE get_precision(precision p) {
-        if (p == precision::SINGLE) {
-            return DFTI_SINGLE;
-        }
-        else {
-            return DFTI_DOUBLE;
-        }
-    }
+    DFT_ERROR status;
+    mklcpu_desc_t device_handle = nullptr;
+    sycl::buffer<mklcpu_desc_t, 1> handle_buffer{ &device_handle, sycl::range<1>{ 1 } };
```

Comment:
If we have a buffer for this, why declare `device_handle` separately, instead of always accessing it via the buffer? Can be constucted using the constructor `buffer(const range<Dimensions>& bufferRange, const property_list& propList = {})`

Reply:
get_handle() returns the class member variable `device_handle` so thats why its needed to be declared separately.